cmd/src/update.go

```diff
@@ -0,0 +1,156 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+
+	"github.com/sourcegraph/src-cli/internal/api"
+
+	"github.com/tj/go-update"
+	"github.com/tj/go-update/progress"
+	"github.com/tj/go-update/stores/github"
+)
+
+const (
+	// PrivateFileMode grants owner to read/write a file.
+	PrivateFileMode = 0600
+)
+
+func init() {
+	usage := `
+Examples:
+
+	Update and replace src-cli to the recommended version:
+
+		$ src update
+`
+
+	flagSet := flag.NewFlagSet("update", flag.ExitOnError)
+
+	var apiFlags = api.NewFlags(flagSet)
+
+	handler := func(args []string) error {
+		currentDirectory, err := os.Getwd()
+		if err != nil {
+			return err
+		}
+		err = isDirWritable(currentDirectory)
+		if err != nil {
+			runWithElevatedPrivilege()
+			return nil
+		}
+
+		fmt.Printf("Current version: %s\n", buildTag)
+
+		client := cfg.apiClient(apiFlags, flagSet.Output())
+		recommendedVersion, err := getRecommendedVersion(context.Background(), client)
+		if err != nil {
+			return err
+		}
+		if recommendedVersion == "" {
```

Comment:
These lines are exactly the same as in `cmd/src/version.go` - maybe put them in a `getCurrentAndRecommendVersion` function?